src/internal/components/file-token-group/interfaces.ts

```diff
@@ -0,0 +1,98 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import { BaseComponentProps } from '../../base-component';
+import { NonCancelableEventHandler } from '../../events';
+
+export interface FileTokenGroupProps extends BaseComponentProps {
+  /**
+   * Show file size in the token. Use `i18nStrings.formatFileSize` to customize it.
+   */
+  showFileSize?: boolean;
+
+  /**
+   * Show file last modified timestamp in the token. Use `i18nStrings.formatFileLastModified` to customize it.
+   */
+  showFileLastModified?: boolean;
+
+  /**
+   * Show file thumbnail in the token. Only supported for images.
+   */
+  showFileThumbnail?: boolean;
+  /**
+   *  Called when the user clicks on the dismiss button. The token won't be automatically removed.
+   *  Make sure that you add a listener to this event to update your application state.
+   */
+  onDismiss: NonCancelableEventHandler<FileTokenGroupProps.DismissDetail>;
+
+  /**
+   * Specifies the maximum number of displayed tokens. If the property isn't set, all of the tokens are displayed.
+   */
+  limit?: number;
+  /**
+   * Specifies the direction in which tokens are aligned (`horizontal | vertical`).
+   */
+  alignment?: FileTokenGroupProps.Alignment;
+
+  /**
+   *
+   * An array of objects representing token items. Each token has the following properties:
+   *
+   * - `file` (string) - File value.
+   * - `disabled` [boolean] - (Optional) Determines whether the token is disabled.
+   * - `loading` (boolean) - (Optional) Determine whether the token is loading.
+   * - `errorText` (string) - (Optional) Text that displays as a validation error message.
+   * - `warningText` (string) - (Optional) - Text that displays as a validation warning message.
+   */
+  items: ReadonlyArray<FileTokenGroupProps.Item>;
+  /**
+   * Adds an `aria-label` to the "Show fewer" button.
+   * Use to assign unique labels when there are multiple file token groups with the same `limitShowFewer` label on one page.
+   */
+  limitShowFewerAriaLabel?: string;
+  /**
+   * Adds an `aria-label` to the "Show more" button.
+   * Use to assign unique labels when there are multiple file token groups with the same `limitShowMore` label on one page.
+   */
+  limitShowMoreAriaLabel?: string;
+  /**
+   * Specifies if the control is read-only, which prevents the
+   * user from modifying the value. A read-only control is still focusable.
+   */
+  readOnly?: boolean;
```

Comment:
Good catch, I decided to remove the disabled property for now since we don't have a use case. I'll remove that property. As an FYI, the reason one is on the group and one on the token is consistent with the [token group](https://cloudscape.aws.dev/components/token-group?tabId=playground&example=read-only-state&props=N4Ig9gdgIglgzgW3nEAuEAKYACAJgUwBcBDGAG1WxxkPwQEkICAPbAX3YEpsBeAPirY4RerQRwMAbQB0smnTjS4ZGAGN8GAAwAabPIZN8zTrtnT9i5Wo37GLbAGpsARk4BdTgG52IbSAtoIJKCZMQARvgUADogonQuMdpRENipePBIcHAAMuGRqDEASnRgAG74emIJIOxJKSF50bFVAEyJyWnpiMi5EU3FCGUV%2Bthtvl3hZPi4lIQATgCuFWx1qTihfQXN8QDM7fWpuBk9jVsDQ5W7MezYbr4gxCoA5hAI%2BBCEgQAWYHMwAF6QEhke5HOCTfAAeQWtDmAAViLgjhAnmgAGaPYR%2BOb4RGQiBkACeaHmSzYQA).